keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -9,105 +9,201 @@
   - [Non-Goals](#non-goals)
 - [Proposal](#proposal)
   - [Preserve existing behavior](#preserve-existing-behavior)
-  - [Updates to API server token generation](#updates-to-api-server-token-generation)
   - [New API](#new-api)
-  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Plugins] (#plugins)
+  - [Updates for token generation](#updates-for-token-generation)
+  - [Updates for supported public keys](#updates-for-supported-public-keys)
+  - [Notes/Constraints/Caveats](#notesconstraintscaveats)
   - [Risks and Mitigations](#risks-and-mitigations)
-- [Graduation Criteria](#graduation-criteria)
+- [Design details](design-details)
+  - [ExternalJWTSigner RPC](#externaljwtsigner-rpc)
+  - [Test Plan](#test-plan)
+	  - [Prerequisite testing updates](#prerequisite-testing-updates)
+	  - [Unit tests](#unit-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade/Downgrade Strategy](#upgradedowngrade-strategy)
+  - [Scalability](#scalability)
 - [Implementation History](#implementation-history)
 <!-- /toc -->
 
 ## Summary
 
-The Kubernetes API server has always read service account keys from disk as the process starts, and kept them in memory for the duration of the server's lifetime. As the API server can now verify and issue projected volume tokens, it would be advantageous to support external signing and verifying of token data over an API, as well as reading public keys from an API.
+Service account keys are used by kube-apiserver for JWT signing and authentication. Keys are stored on the disk and are loaded by kube-apiserver during process start time. Key management for service account keys is entirely within the cluster and is less flexible. This KEP proposes updates that will allow Kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes).  
```

Comment:
nit:

> "... and is less flexible." 

Less flexible than what? It might be better to either compare it to something, or just state what it can't/doesn't do (ex: keys are exportable)

Reply:
reworded to `...is less flexible in that the signing keys remain same for the lifetime of the process`